common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -615,6 +625,162 @@ public boolean verify(@NonNull final SigningAlgorithm alg,
         return false;
     }
 
+    @Override
+    public String encrypt(@NonNull final Cipher cipher,
+                          @NonNull final String plaintext) throws ClientException {
+        final String methodName = ":encrypt";
+        final String errCode;
+        final Exception exception;
+
+        try {
+            // Load our key material
+            final KeyStore.PrivateKeyEntry privateKeyEntry = (KeyStore.PrivateKeyEntry)
+                    mKeyStore.getEntry(mKeyAlias, null);
+
+            // Get a ref to our public key
+            final PublicKey publicKey = privateKeyEntry.getCertificate().getPublicKey();
+
+            // Init our Cipher
+            final javax.crypto.Cipher input = javax.crypto.Cipher.getInstance(cipher.toString());
+            input.init(javax.crypto.Cipher.ENCRYPT_MODE, publicKey);
+
+            // Declare an OutputStream to hold our encrypted data
+            final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+            // Wrap it in our CipherOutputStream, write the contents...
+            final OutputStream cipherOutputStream = new CipherOutputStream(outputStream, input);
+            cipherOutputStream.write(plaintext.getBytes(ENCODING_UTF8));
+            cipherOutputStream.close();
+
+            // Flatten our OutputStream to an array
+            byte[] encryptedBase64Data = outputStream.toByteArray();
+
+            // Base64 encode to stringify
+            return Base64.encodeToString(encryptedBase64Data, Base64.DEFAULT);
```

Comment:
Similar to my other comments, you may be able to use a Base64OutputStream here.

Reply:
Applied in 02668e90e3683ef1ab9c0ee843f7a1457a2e0c8b

Kind've breaking-even syntactically. This may be a bit more idiomatic

Reply:
If we go to min API 19, then we get try-with-resources, where you can:

try(Stream one = makeStream();
     Stream two = makeAnotherStream(one)) {
 do things with streams 
} 

No closing, nothing else.

Can we add multiDexEnabled true to our code, or is there a ProGuard issue with that?